File: sw.java
// import java.util.HashSet;
// class sw{
//     public static void main(String[] args) {
//         HashSet<String> hs = new HashSet<>();
//         String s = "abacddcbkbb";
//         int k = 4;
//         String sub = "";
//         for(int i = 0;i < k;i++){
//             sub += s.charAt(i);
//         }
//         hs.add(sub);
//         System.out.println(hs);
        
//     }
// }
//  current_sum - l[i-k] + l[i]


import java.util.HashMap;
import java.util.Map;
class sw{
    public static void main(String[] args) {
        String s = "abacddcbkbb";
        int k = 4;
        int i = 0, j = 0;
        Map<Character, Integer> map = new HashMap<>();
        while (j < s.length()) {
            map.put(s.charAt(j), map.getOrDefault(s.charAt(j), 0) + 1);
            if (j - i + 1 < k) {
                j++;
            } else if (j - i + 1 == k) {
                if (map.size() == k) {
                    System.out.println(s.substring(i, j + 1));
                }
                map.put(s.charAt(i), map.get(s.charAt(i)) - 1);
                if (map.get(s.charAt(i)) == 0) {
                    map.remove(s.charAt(i));
                }
                i++;
                j++;
            }
        }
    }
}

File: sw.py
# l = [5,3,1,6,7,8,4,2,17,3]
# sl = []
# for i in range(0,len(l)-2):
#     sum = 0
#     for j in range(i,i+3):
#         sum += l[j]
#     sl.append(sum)
# print(sl)

# l = [5,3,1,6,7,8,4,2,17,3]
# k = 3
# max_sum = 0
# current_sum = 0
# for i in range(0,k):
#     current_sum += l[i]
# max_sum = current_sum
# for i in range(k,len(l)):
#     current_sum = current_sum - l[i-k] + l[i]
#     if current_sum > max_sum:
#         max_sum = current_sum
# print(max_sum)

# s = "abacddcbkbb"
# k = 4
# u = []
# for i in range(len(s)-k+1):
#     w = s[i:i+k]
#     if len((set(w))) == k:
#         u.append(w)
# print(u)

